### PR TITLE
print test deck

### PR DESCRIPTION
**Pull request by benadida**
I need to add something that waits for all HMPBs to be rendered before showing the print button (cause pagedjs), but it is otherwise working very well.

Now fixed for QR code creation. The only remaining problem is that the content is being created by pagedjs asynchronously, and it would be nice to disable the print button until that's done. I've tried using `afterRendered` in `HandMarkedPaperBallot`, but I'm having trouble figuring out how to update state of the print button without re-rendering all of the HMPBs a second time. Need some React ninja tricks. Help @beausmith you're my only hope.

Still, this is working in every other way.

### CODE REVIEW

**Comment by benadida** [comment]
ok @beausmith ready for review now with memoized component to prevent re-rendering when print button is enabled.

File: client/src/screens/TallyScreen.tsx [hunk 1]
```diff
@@ -392,6 +392,9 @@ const TallyScreen = () => {
             <HorizontalRule />
             <h2>Pre-Election Features</h2>
             <p>
+              <LinkButton to={routerPaths.printTestDecks}>
+                Print Test Decks
+              </LinkButton>
```
Comment (beausmith) [inline client/src/screens/TallyScreen.tsx:397]: Need to force a space here such that the buttons don't touch in the UI.
```suggestion
              </LinkButton>{' '}
```

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 2]
```diff
@@ -215,7 +207,7 @@ class PostRenderBallotProcessor extends Handler {
               isLiveMode: isLiveMode === 'true',
               precinctId,
               ballotStyleId,
-              pageNumber: parseInt(pageNumber, 10),
+              pageNumber: parseInt(pageNumber || '', 10),
```
[the remark this answers is not in the document]
Comment (benadida) [reply client/src/components/HandMarkedPaperBallot.tsx:210]: I believe it was necessary cause nullable var, but I can check if removing it works.

File: client/src/screens/PrintTestDeckScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  id: '',
+  name: 'All Precincts',
+}
+
+type OnAllRenderedFunctionType = () => void
+
+interface TestDeckBallotsParams {
+  election: Election
+  precinct: Precinct
+  onAllRendered: OnAllRenderedFunctionType
```
[the remark this answers is not in the document]
Comment (benadida) [reply client/src/screens/PrintTestDeckScreen.tsx:39]: leftover from when I was trying to do it in the function signature itself, happy to simplify.

File: client/src/components/HandMarkedPaperBallot.tsx [hunk 4]
```diff
@@ -181,24 +177,20 @@ class PostRenderBallotProcessor extends Handler {
...
-            dataset: {
-              pageNumber: `${pages.length + 1}`,
-            },
-          },
-          id: `page-${pages.length + 1}`,
-        })
+
+        const newPage = { ...pages[pages.length - 1] }
+        newPage.element.dataset.pageNumber = `${pages.length + 1}`
+        newPage.element.dataset.id = `page-${pages.length + 1}`
```
[the remark this answers is not in the document]
Comment (benadida) [reply client/src/components/HandMarkedPaperBallot.tsx:183]: I wouldn't have changed anything about your code here if I wasn't using `element` for all of its DOM goodness, thus needing the real type in Typescript, thus triggering a complaint about `dataset` if I didn't do it this way :)

File: client/src/components/HandMarkedPaperBallot.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (benadida) [reply client/src/components/HandMarkedPaperBallot.tsx:210]: yep, necessary cause it is possibly `undefined`. I'll assert it's not null.

**Pull request merged**